packages/client/.eslintrc.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  extends: [
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:react/recommended",
+    "plugin:prettier/recommended",
+    "prettier/@typescript-eslint"
+  ],
+  plugins: ["@typescript-eslint", "prettier", "react"],
+  env: { node: true, es6: true },
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    sourceType: "module",
+    project: "./tsconfig.json",
+  },
+  rules: {
+    // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+    // e.g. "@typescript-eslint/explicit-function-return-type": "off",
+    "react/jsx-props-no-spreading": "off",
+    "no-use-before-define": "off",
+    "@typescript-eslint/no-use-before-define": "off"
+  },
+  settings:  {
+    react:  {
+      version:  'detect',  // Tells eslint-plugin-react to automatically detect the version of React to use
+    },
+  },
+};
```

Comment:
These might not be working. See [link](https://create-react-app.dev/docs/setting-up-your-editor#experimental-extending-the-eslint-config). I think we have to update our .env file if we wanna extend the eslint config. It is also recommended to include the Create-React-app's base config.

I thought about doing this initially as 90% of the tutorials out there are doing this exact same thing haha. But I was pretty happy with the Create-React-app's base eslint config and introducing more rules might torture us (PTSD from 310 lol), so I ended up not adding anything. Their [base config](https://github.com/facebook/create-react-app/blob/master/packages/eslint-config-react-app/index.js) does not enforce any styling and are only used to[ highlight potential error](https://www.reddit.com/r/reactjs/comments/6p3398/totally_lost_on_createreactapp_with/dkmctwi/?utm_source=share&utm_medium=web2x). On the other hand, stricter rules does help us better understand and factor our code, so I am down with that too 😛

Reply:
Holy I missed the whole set up LOL. Lemme just go back to base config... 😅 I was just trying to make server and client linting as consistent as possible but I don't think that's absolutely needed

Reply:
Haha if consistency is the goal, we could install this [package](https://www.npmjs.com/package/eslint-config-react-app) on our server

Reply:
😆 I am willing compromise